docs/guide/declarative_containers.md

```diff
@@ -0,0 +1,111 @@
+# Declarative Environment Configuration
```

Comment:
I find "environment" a bit vague. Since we're building container images, let's call it that.